core/src/main/java/org/apache/carbondata/core/scan/complextypes/ComplexQueryType.java

```diff
@@ -29,6 +32,8 @@
 
   protected int blockIndex;
 
+  protected Map<Integer, DimensionColumnPage> decodedPagesWithPageNumber = new HashMap<>();
```

Comment:
I think this Map will be better to cache the 'byte[] data',   because getChunkData also takes time

Reply:
This map is to hold to DimensionColumnPage as we are decoding the same page again and again. 'byte[] data' for each row is the query result, we need to get that from the decodedPage

Reply:
OK, I didn't mention the pageNumber and rowNumber

> This map is to hold to DimensionColumnPage as we are decoding the same page again and again. 'byte[] data' for each row is the query result, we need to get that from the decodedPage

OK, I didn't mention the pageNumber and rowNumber